File: plum-frontend/src/dto/Category.ts
export enum Category {
    Malicious = 'malicious',
    Automated = 'automated',
    PossiblyAutomated = 'possibly-automated',
    Unclassified = 'unclassified',
}

export const CategoryLabels: { [category in Category]: string } = {
    [Category.Malicious]: 'Malicious',
    [Category.Automated]: 'Automated',
    [Category.PossiblyAutomated]: 'Possibly automated',
    [Category.Unclassified]: 'Unclassified',
};

File: plum-backend/domain/category.go
package domain

type Category struct {
	s string
}

var (
	CategoryMalicious         = Category{"malicious"}
	CategoryAutomated         = Category{"automated"}
	CategoryPossiblyAutomated = Category{"possibly-automated"}
	CategoryUnclassified      = Category{"unclassified"}
)

var Categories = []Category{
	CategoryMalicious,
	CategoryAutomated,
	CategoryPossiblyAutomated,
	CategoryUnclassified,
}

func (c Category) IsZero() bool {
	return c == Category{}
}

func (c Category) String() string {
	return c.s
}
